File: C#高级编程(第7版)源代码/502259 Code CH53/FSharp/Person.fs
namespace Wrox.ProCSharp.Languages

open System

[<Interface>]
type IDisplay =
    abstract member Display : unit -> unit

type Person(firstName0 : string, lastName0 : string) as this = 
    let mutable firstName, lastName = firstName0, lastName0
    
    new () = Person("unknown", "unknown")
    
    member this.FirstName
        with get() = firstName
        and set(value) = firstName <- value
        
    member this.LastName
        with get() = lastName
        and set(value) = lastName <- value
       
    interface IDisplay with
        member this.Display () = Console.WriteLine("{0} {1}", this.FirstName, this.LastName)
        
    override this.ToString () = this.FirstName + " " + this.LastName

File: C#高级编程(第7版)源代码/502259 Code CH53/CSharp/Person.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp
{
    public interface IDisplay
    {
        void Display();
    }

    public class Person
    {
        public Person()
            : this("unknown", "unknown")
        {
        }


        public Person(string firstName, string lastName)
        {
            this.firstName = firstName;
            this.LastName = lastName;
        }


        private string firstName;
        public string FirstName
        {
            get { return firstName; }
            set { firstName = value; }
        }

        public string LastName { get; set; }
	
    }
}
